core/server/master/src/main/java/alluxio/master/file/meta/cross/cluster/LocalMountState.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.file.meta.cross.cluster;
+
+import alluxio.grpc.NetAddress;
+import alluxio.master.file.meta.options.MountInfo;
+import alluxio.proto.journal.CrossCluster.MountList;
+import alluxio.proto.journal.CrossCluster.RemovedMount;
+
+import java.net.InetSocketAddress;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+
+/**
+ * Tracks the state of the local mounts for publishing them to other clusters.
+ * Will only publish information about cross cluster mounts that are not read only.
+ */
+public class LocalMountState {
```

Comment:
These are the functions that the MountTable will call, which will then update the CrossClusterMaster service